src/store/CurrentSpotStore.js

```diff
@@ -0,0 +1,51 @@
+import { action, observable, when } from 'mobx';
+
+class CurrectSpotStore {
+    constructor(mainStore) {
+        this.mainStore = mainStore;
+        when(() => this.context, this.onContextReady);
+    }
+
+    @observable top = 0;
+    @observable left = 0;
+    @observable show = false;
+
+    get context() { return this.mainStore.chart.context; }
+    get stx() { return this.context.stx; }
+    onContextReady = () => {
+        const self = this;
+        this.stx.append('draw', function () {
```

Comment:
there is not much code now in smartcharts that uses function context to retrieve its members (`this` in this function would be the chart engine). I'd say from a design standpoint it's easier to read. Consider using:

`this.stx.append('draw', this.updateSpot)`

where `updateSpot` is action bound.